### PR TITLE
Also implement Jacobians for MappingFEField.

### PR DESCRIPTION
**Pull request by kronbichler**
This commit restructures the implementation of mapping_fe_field.cc similarly to what has recently done by @bangerth in mapping_q1.cc. @luca-heltai please check that I did not miss anything when making this change.

(This patch passes the MappingFEField tests of the test suite.)

With this change, MappingFEField does now also fill Jacobians, inverse Jacobians, and Jacobian gradients on cells and faces. The test jacobians_face_fe_field is constructed to generate the exact same output as jacobians_face, but using MappingFEField with FESystem(FE_Q(QGaussLobatto(4)) instead of MappingQ(3).

### CODE REVIEW

File: source/fe/mapping_fe_field.cc [hunk 1]
```diff
@@ -370,6 +370,349 @@ MappingFEField<dim,spacedim,VECTOR,DH>::get_subface_data (const UpdateFlags upda
...
+     * Update the Hessian of the transformation from unit to real cell, the
+     * Jacobian gradients.
+     *
+     * Skip the computation if possible as indicated by the first argument.
+     */
+    template <int dim, int spacedim>
+    void
+    maybe_update_jacobian_grads (const CellSimilarity::Similarity              cell_similarity,
+                                 const typename dealii::QProjector<dim>::DataSetDescriptor data_set,
+                                 const typename dealii::MappingFEField<dim,spacedim>::InternalData &data,
```
Comment (bangerth) [inline source/fe/mapping_fe_field.cc]: I don't see you use cell_similarity in this function. Am I blind?

**Comment by bangerth** [comment]
This looks good to me (up to the one question I had), but I'll let @luca-heltai look and merge.

This probably counts as related to #1198 in some way as well.

File: source/fe/mapping_fe_field.cc (hunk 1 above)
Comment (kronbichler) [inline source/fe/mapping_fe_field.cc]: Good remark, I'd forgotten to use cell_similarity here. Thanks.

**Comment by bangerth** [comment]
@luca-heltai ?

**Comment by kronbichler** [comment]
Being an Italian myself, I'd assume that @luca-heltai is on holiday this week. If we have patches depending on this one, I'd say we can merge because it passes the MappingFEField tests for me. But I have no pressing follow-up work myself so I'm fine with waiting until @luca-heltai is available again... :)

**Comment by bangerth** [comment]
Let's not worry about it then -- pending patches are always a problem. @luca-heltai may of course comment on a patch that's already been committed and I'm sure you'd still address his concerns :-)

**Pull request merged**